packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -74,9 +77,8 @@ function preload(
       // @ts-ignore
       seen[dep] = true
       const isCss = dep.endsWith('.css')
-      const cssSelector = isCss ? '[rel="stylesheet"]' : ''
       // @ts-ignore check if the file is already preloaded by SSR markup
-      if (document.querySelector(`link[href="${dep}"]${cssSelector}`)) {
+      if (documentLinkUrls.includes(dep)) {
```

Comment:
This now skips checking `[rel="stylesheet"]` if `isCss` is true, which would regress https://github.com/vitejs/vite/pull/1734. We should probably check the `href` and `rel="stylsheet"` too for css.